### ISSUE TITLE
ActiveIpset failed because ipset already existed

### ISSUE THREAD

**Issue by fasaxc** [Metaswitch/calico#436]
@nbartos hit this during his FV testing:

```
21491-Apr 21 16:15:30.593386 10.41.0.18 err calico-felix: _finish_msg_batch failed.
21492-Traceback (most recent call last):
21493-  File "/usr/lib64/python2.7/site-packages/calico/felix/actor.py", line 248, in _step
21494-  File "/usr/lib64/python2.7/site-packages/calico/felix/ipsets.py", line 345, in _finish_msg_batch
21495-  File "/usr/lib64/python2.7/site-packages/calico/felix/ipsets.py", line 388, in _sync_to_ipset
21496-  File "/usr/lib64/python2.7/site-packages/calico/felix/futils.py", line 99, in check_call
21497-FailedSystemCall: Failed system call (retcode : 1, args : ('ipset', 'restore', '-file', '/tmp/tmpEx166Z'))
21498-  stdout  : 
21499:  stderr  : ipset v6.21.1: Error in line 1: Set cannot be created: set with the same name already exists
```
Since we check whether the ipset exists in advance, perhaps there's a bug in the checking code that makes it assume there's no ipset when in fact there is.

Is there some way to make the ipset operations idempotent?  How about passing "-exist" on the create and always flushing after that?

**Comment by nbartos** [comment]
+1 for idempotent

**Comment by plwhite** [comment]
Can you provide a reference to that atomicity? I believe that ipset restore is NOT atomic, and so this fix is not correct.

**Comment by plwhite** [comment]
I have reproed this, and will put together a fix. It's caused by "set_exists" being calculated before the previous actor exits, in a window where the ipset is used, goes out of scope, and then gets recreated.

**Comment by plwhite** [comment]
We've had some going round in circles on this one. PR #443 fixed it, but so does the latest iteration of PR #437. I've closed the first, and the second is pending review markups and merge.

**Comment by fasaxc** [comment]
Fixed by #437.
